http/src/main/java/io/micronaut/http/binary/BinaryTypeConfiguration.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright 2017-2023 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.http.binary;
+
+import io.micronaut.context.annotation.ConfigurationProperties;
+import io.micronaut.core.annotation.NonNull;
+import io.micronaut.core.util.ArgumentUtils;
+import io.micronaut.http.MediaType;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * Allows checking for MediaTypes that should be considered binary.
+ * This is used for example to determine whether to mime encode an AWS lambda response message (where the response body is a String).
+ *
+ * @author Tim Yates
+ * @since 4.0.0
+ */
+@ConfigurationProperties(BinaryTypeConfiguration.PREFIX)
+public class BinaryTypeConfiguration {
+
+    static final String PREFIX = "micronaut.http.binary-types";
+
+    private static final Set<String> DEFAULT_BINARY_TYPES = Set.of(
+        MediaType.APPLICATION_OCTET_STREAM,
+        MediaType.IMAGE_JPEG,
+        MediaType.IMAGE_PNG,
+        MediaType.IMAGE_GIF,
+        "application/zip"
+    );
+
+    private boolean useDefaultBinaryTypes = true;
+
+    @NonNull
+    private List<String> additionalTypes = new ArrayList<>();
+
+    /**
+     * If this is false then calls to {@link #isMediaTypeBinary(String)} will only check the additional types, and ignore the defaults.
+     *
+     * @return Whether to use the default binary types
+     */
+    public boolean isUseDefaults() {
+        return useDefaultBinaryTypes;
+    }
+
+    /**
+     * Sets whether to use the default binary types.
```

Comment:
Tell the user in the javadocs which shows in the Configuration reference documentation which are the default binary types.

Reply:
Done in https://github.com/micronaut-projects/micronaut-core/pull/9530/commits/06e7c9f1950a219e0ff59023b84b23467562b186